Sources/XMTP/Contacts.swift

```diff
@@ -6,6 +6,98 @@
 //
 
 import Foundation
+import XMTPRust
+
+public enum AllowState: String, Codable {
+	case allowed, blocked, unknown
+}
+
+struct AllowListEntry: Codable, Hashable {
+	enum EntryType: String, Codable {
+		case address
+	}
+
+	static func address(_ address: String, type: AllowState = .unknown) -> AllowListEntry {
+		AllowListEntry(value: address, entryType: .address, permissionType: type)
+	}
+
+	var value: String
+	var entryType: EntryType
+	var permissionType: AllowState
+
+	var key: String {
+		"\(entryType)-\(value)"
+	}
+}
+
+class AllowList {
+	var entries: [String: AllowState] = [:]
+
+	static func load(from client: Client) async throws -> AllowList {
+		let publicKey = client.privateKeyBundleV1.identityKey.publicKey.secp256K1Uncompressed.bytes
+		let privateKey = client.privateKeyBundleV1.identityKey.secp256K1.bytes
+
+		let identifier = try XMTPRust.generate_private_preferences_topic_identifier(RustVec(privateKey)).toString()
+		let envelopes = try await client.query(topic: .allowList(identifier))
+		let allowList = AllowList()
+
+		for envelope in envelopes.envelopes {
+
+
+			let payload = try XMTPRust.ecies_decrypt_k256_sha3_256(
+				RustVec(publicKey),
+				RustVec(privateKey),
+				RustVec(envelope.message)
+			)
+
+			let entry = try JSONDecoder().decode(AllowListEntry.self, from: Data(payload))
+
+			allowList.entries[entry.key] = entry.permissionType
+		}
+
+		return allowList
+	}
+
+	static func publish(entry: AllowListEntry, to client: Client) async throws {
+		let payload = try JSONEncoder().encode(entry)
+
+		let publicKey = client.privateKeyBundleV1.identityKey.publicKey.secp256K1Uncompressed.bytes
+		let privateKey = client.privateKeyBundleV1.identityKey.secp256K1.bytes
+		let identifier = try XMTPRust.generate_private_preferences_topic_identifier(RustVec(privateKey)).toString()
```

Comment:
This is never going to change, so we can probably do this once on instantiation of the class instead of on each operation.